RecoLocalCalo/HcalRecAlgos/src/HcalDeterministicFit.cc

```diff
@@ -139,7 +149,17 @@ void HcalDeterministicFit::phase1Apply(const HBHEChannelInfo& channelData,
 
     get205Frac(-tsShift5,-tsShift5+tsWidth,i5);
     get205Frac(-tsShift5+tsWidth,-tsShift5+tsWidth*2,n5);
-  } else {
+  }else if(channelData.hasTimeInfo() && channelData.recoShape()==206) {
+    get206Frac(-tsShift3,-tsShift3+tsWidth,i3);
+    get206Frac(-tsShift3+tsWidth,-tsShift3+tsWidth*2,n3);
+    get206Frac(-tsShift3+tsWidth*2,-tsShift3+tsWidth*3,nn3);
+
+    get206Frac(-tsShift4,-tsShift4+tsWidth,i4);
+    get206Frac(-tsShift4+tsWidth,-tsShift4+tsWidth*2,n4);
+
+    get206Frac(-tsShift5,-tsShift5+tsWidth,i5);
+    get206Frac(-tsShift5+tsWidth,-tsShift5+tsWidth*2,n5);
+  }else {
     getLandauFrac(-tsShift3,-tsShift3+tsWidth,i3);
```

Comment:
these 3 copy-pasted pieces could be easily refactored by introducing ```getFrac(float tStart, float tEnd, float &, enumType)``` or  by picking the right function via a function pointer.